prv.js

```diff
@@ -28,75 +28,116 @@ ensureDirs()
 // keystore: return a JSON-encoded keystore for ADDRESS's key at KEYPATH (protected by KEYPASS)
 // sign:     return signature (hexstring) for MESSAGE using ADDRESS's key at KEYPATH
 
-const inputLines = []
-for await (const line of createInterface({input: process.stdin}))
-  inputLines.push(line)
-
-const readLine = (variable) => {
-  const prefix = `${variable} = `
-  const line = inputLines.shift()
-  if (!(line && line.startsWith(prefix)))
-    throw new Error(`missing ${variable}`)
-  return line.slice(prefix.length)
-}
-
-const chainId = readLine('CHAINID')
-if (!(chainId in chainIds))
-  throw new Error('invalid chainId')
-
-const address = readLine('ADDRESS')
-if (!addressRegExp.test(address))
-  throw new Error('invalid address')
-
-const chainPath = `${workDir}/${chainId}`
-const seedPath = `${chainPath}/${address}`
-
-const command = readLine('COMMAND')
-
-if (command == 'generate') {
-  if (existsSync(seedPath))
-    console.log('exists')
-  else {
-    mkdirSync(chainPath, {recursive: true})
-    writeFileSync(seedPath, randomSeed(), {flag: 'wx'})
-    gitCheck(['add', seedPath], workDir, '', 'failed to add seed')
-    gitCheck(['diff', '--staged', '--name-status'], workDir,
-      output => (
-        !output.trimEnd().includes('\n') &&
-        output.trimEnd().split(/\s+/).join() == `A,${chainId}/${address}`
-      ),
-      'unexpected diff adding seed'
-    )
-    gitPush(address, workDir)
-    console.log('created')
+const [HOST, PORT] = process.env.PRV_HOST.split(':')
+
+createServer((stream) => {
+  console.log("Got new connection")
+
+  const inputLines = []
+  let buffer = ''
+
+  stream.on('data', function (data) {
+    buffer += data.toString();
+  })
+
+  stream.on('end', function () {
+    buffer.split('\n').forEach(line => {
```

Comment:
It would probably be nice to push into `inputLines` on `data` event rather than all at the end... but maybe not super high priority since this endpoint isn't open to public anyway

Reply:
I actually started out that way, but since the data is sent as a stream, you don't know if the data is fully sent yet in the `data` event. In theory, it could be a partial line right?

Reply:
When you see a newline you know what you have before that is a full line. You still need to keep `buffer` but it doesn't grow so long.